utils/lldp.go

```diff
@@ -0,0 +1,179 @@
+/***
+Copyright 2015 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"errors"
+	"os/exec"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Implement utilities to fetch lldp information aobut connecting neighbors.
+// The lldp information can be used in non cloud environment to automate
+// networking configuration on the physical devices
+
+// LLDPNeighbor lists the neighbor attributes as learned by the host
+type LLDPNeighbor struct {
+	SystemName        string
+	SystemDescription string
+	PortID            string
+	PortDesc          string
+	CiscoACIPodID     string
+	CiscoACINodeID    string
+}
+
+// FetchLLDPInfo reads the lldp information using lldptool
+// A native go lib could be done but it requires that lldpad daemon is
+// aceessible natively on the lldp ipc path
+func FetchLLDPInfo(ifname string) (*LLDPNeighbor, error) {
+	lldp := &LLDPNeighbor{}
+
+	output, err := exec.Command("docker", "exec", "lldpad", "lldptool",
+		"-n", "-t", "-i", ifname).CombinedOutput()
+	if err != nil {
+		log.Errorf("Error reading lldp information. Error: %s Output: \n%s\n", err, output)
```

Comment:
since this utility depends on running lldp running in a container, it might also be good to include `Dockerfile` somewhere (`scripts/utils/lldp`?). And perhaps also add checks for container is running and return a more specific (user friendly) error than a more cryptic `docker exec` failure.